src/planner/binder/statement/bind_create.cpp

```diff
@@ -615,6 +618,23 @@ BoundStatement Binder::Bind(CreateStatement &stmt) {
 		}
 		break;
 	}
+	case CatalogType::DATABASE_ENTRY: {
+		// not supported in DuckDB but allow replacement scan
+		auto &base = (CreateDatabaseInfo &)*stmt.info;
+		string extension_name = base.extension_name;
+		string database_name = base.name;
+		auto &config = DBConfig::GetConfig(context);
+		for (auto &extension : config.create_database_extensions) {
+			auto create_database_function_ref =
+			    extension.function(context, extension_name, database_name, extension.data.get());
+			if (create_database_function_ref) {
+				auto bound_create_database_func = Bind(*create_database_function_ref);
+				result.plan = CreatePlan(*bound_create_database_func);
+				break;
+			}
+		}
+		throw NotImplementedException("CREATE DATABASE not supported in DuckDB yet!");
```

Comment:
Won't this always get reached when we enter this case statement?
Within the for loop you use a break, but that does not extend to the outer scope, only the for loop itself

Reply:
oops, thanks for catching this. fixed [here](9a6c464acf0cdc5abe1a26877432c547aaf1d11e).